From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Radek=20Barto=C5=88?= <radek.barton@microsoft.com>
Date: Tue, 3 Dec 2024 14:53:45 +0100
Subject: [PATCH 17/29] 
 0005-Cygwin-define-RTS_CONTROL_ENABLE-and-DTR_CONTROL_ENA.patch

(cherry picked from commit 9356f55388b8ce04c6621caec5996c0cfef5abb9)
---
 gcc/ada/s-oscons-tmplt.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gcc/ada/s-oscons-tmplt.c b/gcc/ada/s-oscons-tmplt.c
index cb2e05f6aae..1f33e902942 100644
--- a/gcc/ada/s-oscons-tmplt.c
+++ b/gcc/ada/s-oscons-tmplt.c
@@ -1041,6 +1041,11 @@ CND(VEOL2, "Alternative EOL")
 
 #endif /* HAVE_TERMIOS */
 
+#if defined(__CYGWIN__)
+#define RTS_CONTROL_ENABLE 0x1
+#define DTR_CONTROL_ENABLE 0x1
+#endif
+
 #if defined(__MINGW32__) || defined(__CYGWIN__)
 CNU(DTR_CONTROL_ENABLE, "Enable DTR flow ctrl")
 CNU(RTS_CONTROL_ENABLE, "Enable RTS flow ctrl")
